File: src/main/java/de/tectoast/emolga/commands/various/RevolutionCommand.java
package de.tectoast.emolga.commands.various;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import net.dv8tion.jda.api.entities.ChannelType;
import net.dv8tion.jda.api.entities.Guild;
import net.dv8tion.jda.api.entities.GuildChannel;
import net.dv8tion.jda.api.entities.Member;
import org.jsolf.JSONArray;
import org.jsolf.JSONObject;

public class RevolutionCommand extends Command {

    public RevolutionCommand() {
        super("revolution", "muhahahahaha", CommandCategory.Various);
        setArgumentTemplate(ArgumentManagerTemplate.builder()
                .add("name", "Name", "Name der Revolution", ArgumentManagerTemplate.Text.any())
                .setExample("!revolution Emolga")
                .build());
        setCustomPermissions(PermissionPreset.CULT);
    }

    @Override
    public void process(GuildCommandEvent e) {
        if (e.getArgsLength() == 0) {
            e.reply("Du musst einen Revolution-Namen angeben!");
            return;
        }
        String name = e.getArg(0);
        Guild g = e.getGuild();
        JSONArray arr = getEmolgaJSON().getJSONArray("activerevolutions");
        boolean isRevo = arr.toList().contains(g.getIdLong());
        JSONObject o = new JSONObject();
        e.getChannel().sendMessage("Möge die **" + name + "-Revolution** beginnen! :D").queue();
        g.loadMembers().onSuccess(list -> {
            for (Member member : list) {
                if (member.isOwner()) continue;
                if (member.getId().equals(e.getJDA().getSelfUser().getId()))
                    member.modifyNickname(name + "leader").queue();
                if (!g.getSelfMember().canInteract(member)) continue;
                if (!isRevo)
                    o.put(member.getId(), member.getEffectiveName());
                member.modifyNickname(name).queue();
            }
            for (GuildChannel gc : g.getChannels()) {
                gc.getManager().setName((gc.getType() == ChannelType.TEXT ? name.toLowerCase() : name) + "-" + gc.getName()).queue();
            }
            if (!isRevo) {
                getEmolgaJSON().getJSONObject("revolutionreset").put(g.getId(), o);
                arr.put(g.getIdLong());
            }
            saveEmolgaJSON();
        });
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/various/RevolutionCommand.kt
package de.tectoast.emolga.commands.various

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.jsolf.JSONObject
import net.dv8tion.jda.api.entities.ChannelType
import net.dv8tion.jda.api.entities.Member
import java.util.*

class RevolutionCommand : Command("revolution", "muhahahahaha", CommandCategory.Various) {
    init {
        argumentTemplate = ArgumentManagerTemplate.builder()
            .add("name", "Name", "Name der Revolution", ArgumentManagerTemplate.Text.any())
            .setExample("!revolution Emolga")
            .build()
        setCustomPermissions(PermissionPreset.CULT)
    }

    override fun process(e: GuildCommandEvent) {
        if (e.argsLength == 0) {
            e.reply("Du musst einen Revolution-Namen angeben!")
            return
        }
        val name = e.getArg(0)
        val g = e.guild
        val arr = emolgaJSON.getJSONArray("activerevolutions")
        val isRevo = arr.toList().contains(g.idLong)
        val o = JSONObject()
        e.textChannel.sendMessage("Möge die **$name-Revolution** beginnen! :D").queue()
        g.loadMembers().onSuccess { list: List<Member> ->
            for (member in list) {
                if (member.isOwner) continue
                if (member.id == e.jda.selfUser.id) member.modifyNickname(name + "leader").queue()
                if (!g.selfMember.canInteract(member)) continue
                if (!isRevo) o.put(member.id, member.effectiveName)
                member.modifyNickname(name).queue()
            }
            for (gc in g.channels) {
                gc.manager.setName((if (gc.type == ChannelType.TEXT) name.lowercase(Locale.getDefault()) else name) + "-" + gc.name)
                    .queue()
            }
            if (!isRevo) {
                emolgaJSON.getJSONObject("revolutionreset").put(g.id, o)
                arr.put(g.idLong)
            }
            saveEmolgaJSON()
        }
    }
}